### PR TITLE
Add support for path names containing special characters

### PR DESCRIPTION
**Pull request by eric-wieser**
This uses the builtin (private) `tkinter._stringify` function, which correctly serializes nested lists of strings no matter what characters they contain.

Also looks like my editor stripped trailing spaces

### CODE REVIEW

**Comment by themperek** [comment]
This makes multiple issues (return '.' for an empty list and some super strange bugs on mac). Will need to find another way probably.

**Comment by eric-wieser** [comment]
>return '.' for an empty string? 
Not sure what you mean by that. This does the right thing on an empty string
```
>>> print(as_tcl_value(''))
{}  # braces are a raw string without interpolation in TCL
```

File: cocotb_test/run.py [hunk 1]
```diff
@@ -243,50 +247,62 @@ def _run_icarus(toplevel, libs_dir, verilog_sources_abs, sim_build_dir, include_
...
+        )
         os.environ["GPI_EXTRA"] = 'fli'
-         
+
     if verilog_sources_abs:
-        do_script += 'vlog -mixedsvvh +define+COCOTB_SIM -sv {INCDIR} {VERILOG_SOURCES}\n'.format(VERILOG_SOURCES=" ".join(verilog_sources_abs), INCDIR=" ".join(inculde_dir_cmd))
-    
+        do_script += 'vlog -mixedsvvh +define+COCOTB_SIM -sv {INCDIR} {VERILOG_SOURCES}\n'.format(
+            VERILOG_SOURCES=as_tcl_value(verilog_sources_abs),
+            INCDIR=as_tcl_value(inculde_dir_cmd)
```
Comment (eric-wieser) [inline cocotb_test/run.py]: Ah crap, these were wrong - they should have been `" ".join(as_tcl_value(v) for v in verilog_sources_abs)`

**Comment by themperek** [comment]
Fixed

**Comment by themperek** [comment]
@eric-wieser Would it be enough to call as_tcl_value at the end on the full script and not multiple times?

**Comment by eric-wieser** [comment]
No, the whole point of as_tcl_value is to escape python data when you inject it into the script. Think of it like a html_escape function - you escape the user name, not the entire webpage.

**Comment by themperek** [comment]
@eric-wieser Thank you!

**Pull request merged**